GameOfLife/GameOfLife/Controllers/InputController.cs

```diff
@@ -467,51 +482,58 @@ public bool CheckInputMultipleGamesMenu(ConsoleKey keyPressed)
-            Console.CursorVisible = true;
-            Console.Clear();
-            while (true)
+            switch (keyPressed)
             {
-                Console.Write(EnterTotalGamesNumberPhrase);
-                userInput = Console.ReadLine();
-                if (int.TryParse(userInput, out var totalNumberOfGames) && totalNumberOfGames > 1 && totalNumberOfGames <= 2000)
-                {
-                    MultipleGames.TotalNumberOfGames = totalNumberOfGames;
-                    break;
-                }
-                else
-                {
-                    Console.WriteLine(WrongInputPhrase);
-                }
-            }
+                case ConsoleKey.D1:
+                    MultipleGames.Length = 10;
+                    MultipleGames.Width = 10;
+                    MultipleGames.NumberOfGamesToBeDisplayed = 24;
+                    return MultipleGames;
 
-            while (true)
-            {
-                Console.Write(EnterLengthMultipleGamesPhrase);
-                userInput = Console.ReadLine();
-                if (int.TryParse(userInput, out var length) && length >= 3 && length <= 30)
-                {
-                    MultipleGames.Length = length;
-                    break;
-                }
-                else
-                {
-                    Console.WriteLine(WrongInputPhrase);
-                }
+                case ConsoleKey.D2:
+                    MultipleGames.Length = 15;
+                    MultipleGames.Width = 15;
+                    MultipleGames.NumberOfGamesToBeDisplayed = 12;
+                    return MultipleGames;
+
+                case ConsoleKey.D3:
+                    MultipleGames.Length = 20;
+                    MultipleGames.Width = 20;
+                    MultipleGames.NumberOfGamesToBeDisplayed = 6;
+                    return MultipleGames;
+
+                case ConsoleKey.D4:
+                    MultipleGames.Length = 25;
+                    MultipleGames.Width = 25;
+                    MultipleGames.NumberOfGamesToBeDisplayed = 6;
+                    return MultipleGames;
+
+                default:
+                    return null;
             }
+        }
 
+        /// <summary>
+        /// Method to take and process user's input of the number of games and Game Field sizes for the Multiple Games Mode.
+        /// </summary>
+        public MultipleGamesModel EnterNumberOfMultipleGames(MultipleGamesModel multipleGames)
+        {
+            string userInput;
+            MultipleGames = multipleGames;
+            Console.CursorVisible = true;
             while (true)
             {
-                Console.Write(EnterWidthMultipleGamesPhrase);
                 userInput = Console.ReadLine();
-                if (int.TryParse(userInput, out var width) && width >= 3 && width <= 10)
+                if (int.TryParse(userInput, out var totalNumberOfGames) && totalNumberOfGames >= 100 && totalNumberOfGames <= 10000)
```

Comment:
Is there a specific reason, why you require at least 100 games here? For now, you have split most of your logics in single game methods vs multi game methods. What if you could update this logic, and allow user to input from 1 till 1000 games? Then you could use your multi game logic everywhere, and then get rid of the single game logics/methods, as those would be already taken care of. In the end, your logic should be generic enough, that it does not really care, how many games you choose (1, 13, 156, 565, 1000) and can process this amount of games with the same methods/logics, not requiring specific methods if user chooses 1 game vs multiple games.

Reply:
100 just seemed to me like a decent minimal amount of games. But I get your point, will do.

Reply:
United methods that were responsible for Single Game Mode and methods that were responsible for Multiple Games Mode. Removed redundant methods, did some refactoring and restructurizations.